src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Forms/FormsWithFiles.razor

```diff
@@ -0,0 +1,57 @@
+﻿@page "/forms/with-files"
+@using Microsoft.AspNetCore.Components.Forms
+
+<h2>Forms With Files</h2>
+
+<EditForm Model="Model" method="post" OnValidSubmit="DisplaySuccess" FormName="fileForm" enctype="multipart/form-data">
+    <ValidationSummary />
+    <div>
+        <label for="Model.ProfilePicture">
+            Profile Picture:
+            <InputFile @bind-Value="Model.ProfilePicture" name="Model.ProfilePicture" />
+        </label>
+        <ValidationMessage For="() => Model.ProfilePicture" />
+    </div>
+    <div>
+        <label for="Model.Documents">
+            Documents:
+            <InputFile @bind-Value="Model.Documents" multiple name="Model.Documents" />
+        </label>
+        <ValidationMessage For="() => Model.Documents" />
+    </div>
+    <div>
+        <label for="Model.Documents">
+            Header Photo:
+            <InputFile @bind-Value="Model.HeaderPhoto" name="Model.HeaderPhoto" />
```

Comment:
`InputFile` doesn't have any `Value` you can bind to, so this looks like a mistake. I was surprised at first that the compiler even allowed this, but it must follow from the fact that we don't restrict passing parameters to ones that actually exist. It's just unfortunate that the editor highlights `@bind-Anything` in purple as if it's definitely correct.

For these E2E tests, the right move is probably to replace `<InputFile @bind-Value>` with `<input type="file" name="Model.HeaderPhoto">` etc.

Reply:
> InputFile doesn't have any Value you can bind to, so this looks like a mistake. I was surprised at first that the compiler even allowed this, but it must follow from the fact that we don't restrict passing parameters to ones that actually exist. 

Yep, this is correct. The `name` attribute is the only reason this works correctly for posted forms.

> For these E2E tests, the right move is probably to replace <InputFile @bind-Value> with <input type="file" name="Model.HeaderPhoto"> etc.

I'm fine with this. I assume we'll use https://github.com/dotnet/aspnetcore/issues/50728 to track whether `InputFile` should be updated to more closely match `InputBase`.

Reply:
Yes. This was a small oversight on my part.